src/test/java/ArrayIndexedBagEqualsTest.java

```diff
@@ -33,9 +35,58 @@ public void equals_verify_contract() {
         manyC.add(10);
         manyC.add(20);
         manyC.add(30);
+        manyDifferentOrder.add(20);
+        manyDifferentOrder.add(30);
+        manyDifferentOrder.add(10);
         manySortedBag.add(10);
         manySortedBag.add(20);
         manySortedBag.add(30);
+        manySortedBagDifferentOrder.add(30);
+        manySortedBagDifferentOrder.add(10);
+        manySortedBagDifferentOrder.add(20);
+
+        ArrayIndexedBag<Integer> duplicateA = new ArrayIndexedBag<>();
+        ArrayIndexedBag<Integer> duplicateB = new ArrayIndexedBag<>();
+        ArrayIndexedBag<Integer> duplicateC = new ArrayIndexedBag<>(1000);
+        ArrayIndexedBag<Integer> duplicateDifferentOrder = new ArrayIndexedBag<>();
+        ArraySortedBag<Integer> duplicateSortedBag = new ArraySortedBag<>();
+        ArraySortedBag<Integer> duplicateSortedBagDifferentOrder = new ArraySortedBag<>();
+        duplicateA.add(10);
```

Comment:
Tests are quite long, a little hard to follow. Consider defining a private helper method like the following:
```java
private ArrayIndexedBag<T> of(T... ts) {
  ArrayIndexedBag<T> bag = new ArrayIndexedBag<>();
  for (T t : ts) { bag.add(t); }
  return bag;
}
```
And maybe another one for the non-class under test, and for capacities, etc. Using it in the actual test method is much clearer then,
```java
new EqualsTester()
  .addEqualityGroup(of<Integer>(), of<Integer>(), ofCap<Integer>(100), ofSorted<Integer>())
  .addEqualityGroup(of(1), of(1), ofSorted(1))
  .addEqualityGroup(of(1, 2), of(1, 2), ofSorted(1, 2))
```

Intellij et al will warn on `T...` being weird, but it's safe enough. You can also try slapping [@SafeVarargs](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/SafeVarargs.html) on it, but it may fail (I honestly forget).

Reply:
yo, this is nice :man_cook: :kiss: